### PR TITLE
Fix cross-chain payments made to invalid addresses

### PR DESCRIPTION
**Pull request by quantumagi**

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/SourceChain/DepositExtractor.cs [hunk 1]
```diff
@@ -34,10 +34,26 @@ public DepositExtractor(IFederatedPegSettings federatedPegSettings, Network netw
             this.opReturnDataReader = opReturnDataReader;
         }
 
+        private static Dictionary<string, List<IDeposit>> DepositsToInject = new Dictionary<string, List<IDeposit>>()
+        {
+            { "CirrusRegTest", new List<IDeposit> { 
+                new Deposit(0x1 /* Tx of deposit being redone */, DepositRetrievalType.Small, new Money(10, MoneyUnit.BTC), "qZc3WCqj8dipxUau1q18rT6EMBN6LRZ44A", DestinationChain.STRAX, 12500, 0) } }
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/SourceChain/DepositExtractor.cs:40]: What transaction id will be injected here? Is it the one that was originally done incorrectly?

Comment (fassadlr) [reply src/Stratis.Features.FederatedPeg/SourceChain/DepositExtractor.cs:40]: If that is the case... I'm wondering if we shouldn't check whether it actually exists first?

Also, what stops quorum amount of Multisig members from amending the code and injecting deposits as they please?

Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/SourceChain/DepositExtractor.cs:40]: Yes, and you are mirroring my thoughts on this. We could check if it exists as the core principle here is that we should only process deposits that have bad addresses and have not been processed before. However it does seem a bit unnecessary as we would know all those things before adding it to the list.

> Also, what stops quorum amount of Multisig members from amending the code and injecting deposits as they please?

I don't think it really relates to quorum but more our ability to modify the code base. In that case the answer would be the same as for any code change we make. The users vote by downloading and using those code (or not).

**Review by fassadlr — APPROVED** [review]
@quantumagi before we merge we need to build some confidence around this. Can we at least have a unit test to prove that the injected deposit and subsequent withdrawal transaction was created?

Either that or we need to run this on CirrusTest or CirrusRegTest and prove that it works :)

**Comment by quantumagi** [comment]
I can test this on my CirrusRegTest environment.

 We also need to update the code with the list of deposits that we want to recover on CirrusMain.

**Comment by quantumagi** [comment]
Deposit successfully injected:

**Comment by fassadlr** [comment]
@quantumagi these are ones we need to inject:

https://cirrusexplorer.stratisplatform.com/transactions/6179ee3332348948641210e2c9358a41aa8aaf2924d783e52bc4cec47deef495
239CRS should go to `XNrgftud4ExFL7dXEHjPPx3JX22qvFy39v`

https://cirrusexplorer.stratisplatform.com/transactions/b8417bdbe7690609b2fff47d6d8b39bed69993df6656d7e10197c141bdacdd90
640CRS should go to `XCDeD5URnSbExLsuVYu7yufpbmb6KvaX8D`

https://cirrusexplorer.stratisplatform.com/transactions/375ed4f028d3ef32397641a0e307e1705fc3f8ba76fa776ce33fff53f52b0e1c
1649CRS should go to `XCrHzx5AUnNChCj8MP9j82X3Y7gLAtULtj`

**Comment by quantumagi** [comment]
```
            { "CirrusMain", new List<IDeposit> {
                new Deposit(
                    uint256.Parse("6179ee3332348948641210e2c9358a41aa8aaf2924d783e52bc4cec47deef495") /* Tx of deposit being redone */, DepositRetrievalType.Normal, 
                    new Money(239, MoneyUnit.BTC), "XNrgftud4ExFL7dXEHjPPx3JX22qvFy39v", DestinationChain.STRAX, 2500000, 0),
                new Deposit(
                    uint256.Parse("b8417bdbe7690609b2fff47d6d8b39bed69993df6656d7e10197c141bdacdd90") /* Tx of deposit being redone */, DepositRetrievalType.Normal,
                    new Money(640, MoneyUnit.BTC), "XCDeD5URnSbExLsuVYu7yufpbmb6KvaX8D", DestinationChain.STRAX, 2500000, 0),
                new Deposit(
                    uint256.Parse("375ed4f028d3ef32397641a0e307e1705fc3f8ba76fa776ce33fff53f52b0e1c") /* Tx of deposit being redone */, DepositRetrievalType.Large,
                    new Money(1649, MoneyUnit.BTC), "XCrHzx5AUnNChCj8MP9j82X3Y7gLAtULtj", DestinationChain.STRAX, 2500000, 0),
                }
            }
```

**Pull request merged**